### ISSUE TITLE
Questions about replicated vectors in boundaries across postings.

### ISSUE THREAD

**Issue by redeggandsky** [SPFresh/SPFresh#18]
Considering clustering methods such as SPANN, a vector can be inserted into multiple clusters, which is somewhat confusing in this paper.

1. Which approach is used in SPFresh: only inserting the vector into the nearest partition, or sending copies to nearby partitions as suggested in SPANN?
2. If the latter is the case, how should we handle the issue of duplicate vectors during the merging process? If we simply eliminate duplicates, we may end up with fewer and fewer backups of the same vector.

**Comment by Yuming-Xu** [comment]
Thanks for the great question! :)

SPFresh follows the same approach as SPANN: it inserts replicas into nearby partitions.
In my opinion, if two partitions are small enough, merging them can reduce both the cost of traversing the upper-layer SPTAG to locate them and the cost of fetching the vectors inside, since they’re much smaller than 4KB on disk. In such cases, replicating a vector in both partitions won’t negatively impact accuracy. In fact, we’ve seen situations where accuracy improves when merging short partitions, even before inserting any new vectors.

In short, I don’t think it will be a problem if some vectors have fewer backups. If you come across cases where fewer backups do cause issues, I’d be happy to discuss them further. :)
